sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -223,10 +224,10 @@ class Dataset[T] private[sql](
   @transient private[sql] val logicalPlan: LogicalPlan = {
     // For various commands (like DDL) and queries with side effects, we force query execution
     // to happen right away to let these side effects take place eagerly.
-    val plan = queryExecution.analyzed match {
+    val plan = queryExecution.commandCollected match {
```

Comment:
I think we can remove the pattern match and simply write `val plan = queryExecution.commandCollected`